### ISSUE TITLE
Support for schema references with kafka schema registry.

### ISSUE THREAD

**Issue by vipul1409** [AbsaOSS/ABRiS#187]
I tried reading schema information which references other schemas in schema registry it failed with below error.
Code used for loading schema is 
```
AbrisConfig
      .fromSimpleAvro
      .downloadSchemaByLatestVersion
      .andTopicNameStrategy(readTopic)
      .usingSchemaRegistry(DefaultSchemaRegistry)
```

```
21/02/17 06:11:25 INFO SchemaManagerFactory: Configuring new Schema Registry instance of type 'io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient'
Exception in thread "main" com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "references" (class io.confluent.kafka.schemaregistry.client.rest.entities.Schema), not marked as ignorable (4 known properties: "version", "schema", "id", "subject"])
at [Source: sun.net.www.protocol.http.HttpURLConnection$HttpInputStream@4f5c30b1; line: 1, column: 2017] (through reference chain: io.confluent.kafka.schemaregistry.client.rest.entities.Schema["references"])at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:52)
at com.fasterxml.jackson.databind.DeserializationContext.reportUnknownProperty(DeserializationContext.java:839)
at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1045)
at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1352)
at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperties(BeanDeserializerBase.java:1306)
at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:399)
at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1099)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:296)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:133)
at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3736)
at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2803)
at io.confluent.kafka.schemaregistry.client.rest.RestService.sendHttpRequest(RestService.java:221)
at io.confluent.kafka.schemaregistry.client.rest.RestService.httpRequest(RestService.java:265)
at io.confluent.kafka.schemaregistry.client.rest.RestService.getLatestVersion(RestService.java:524)
at io.confluent.kafka.schemaregistry.client.rest.RestService.getLatestVersion(RestService.java:516)
at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.getLatestSchemaMetadata(CachedSchemaRegistryClient.java:275)
at za.co.absa.abris.avro.read.confluent.SchemaManager.getSchemaMetadataBySubjectAndVersion(SchemaManager.scala:64)
at za.co.absa.abris.avro.read.confluent.SchemaManager.getSchemaBySubjectAndVersion(SchemaManager.scala:53)
at za.co.absa.abris.avro.read.confluent.SchemaManager.getSchema(SchemaManager.scala:44)
at za.co.absa.abris.config.FromSchemaDownloadingConfigFragment.usingSchemaRegistry(Config.scala:194)
at za.co.absa.abris.config.FromSchemaDownloadingConfigFragment.usingSchemaRegistry(Config.scala:189)
```

**Comment by cerveada** [comment]
Schema references are supported from confluent 5.5.0
https://www.confluent.io/blog/multiple-event-types-in-the-same-kafka-topic/

Abris is using confluent version 5.3.4.

The newer version was causing compatibility issues with libraries used by Spark (details in #165).

**Comment by vipul1409** [comment]
@cerveada If I am understanding correctly there is no way to use Schema reference in Spark with Abris. Because with 5.3.4 one can't use references and 5.5 will not work with Spark.

**Comment by cerveada** [comment]
yes

**Comment by kevinwallimann** [comment]
@vipul1409 As explained in https://github.com/AbsaOSS/ABRiS/issues/165, the incompatibility issue for Confluent 5.4.0 was introduced by Avro 1.9. We believe the compatibility issue can be resolved with Avro 1.10 which will be supported in Spark 3.2.0 (see https://issues.apache.org/jira/browse/SPARK-27733)
